ios/browser/api/ads/brave_ads.mm

```diff
@@ -252,52 +245,47 @@ - (void)initServiceWithSysInfo:(BraveAdsSysInfo*)sysInfo
     return completion(/*success=*/false);
   }
 
-  [self initDatabase];
-
-  ads = brave_ads::Ads::CreateInstance(*adsClient);
-
   auto cppSysInfo =
       sysInfo ? sysInfo.cppObjPtr : brave_ads::mojom::SysInfo::New();
-  ads->SetSysInfo(std::move(cppSysInfo));
   auto cppBuildChannelInfo = buildChannelInfo
                                  ? buildChannelInfo.cppObjPtr
                                  : brave_ads::mojom::BuildChannelInfo::New();
-  ads->SetBuildChannel(std::move(cppBuildChannelInfo));
-  ads->SetFlags(brave_ads::BuildFlags());
-
   auto cppWalletInfo =
       walletInfo ? walletInfo.cppObjPtr : brave_ads::mojom::WalletInfoPtr();
-  ads->Initialize(std::move(cppWalletInfo),
-                  base::BindOnce(^(const bool success) {
-                    [self periodicallyCheckForAdsResourceUpdates];
-                    [self registerAdsResources];
-                    if (success) {
-                      [self notifyDidInitializeAds];
-                    }
-                    completion(success);
-                    if (!success) {
-                      [self cleanupAds];
-                    }
-                  }));
+
+  ProfileIOS* profile = [self getLastUsedProfile];
+  adsService = brave_ads::AdsServiceFactoryIOS::GetForBrowserState(profile);
+  CHECK(adsService);
+
+  adsService->InitializeAds(
+      base::SysNSStringToUTF8(self.storagePath), *adsClient,
+      std::move(cppSysInfo), std::move(cppBuildChannelInfo),
+      std::move(cppWalletInfo), base::BindOnce(^(const bool success) {
+        [self periodicallyCheckForAdsResourceUpdates];
+        [self registerAdsResources];
```

Comment:
Do we know if there's a reason to register or check for resources when ads fail to initialize? It doesn’t seem like we need to. Also, would it make more sense to switch these two calls, so we’re registering first and then checking for updates after?

Reply:
Great catch! I don't see any reasons for this. Changed the code to run `registerAdsResources` and `periodicallyCheckForAdsResourceUpdates` only if InitializeAds succeeded.
Also added `[self stopComponentUpdaterTimer];` call in `shutdownService`